modules/core/src/main/scala/org/scalasteward/core/sbt/SbtAlg.scala

```diff
@@ -125,6 +128,31 @@ object SbtAlg {
       val sbtDir: F[File] =
         fileAlg.home.map(_ / ".sbt")
 
+      val buildPropertiesFile: F[File] =
+        fileAlg.home.map(_ / "project" / "build.properties")
+
+      final private val sbtVersionRegex = s"build.properties=(.+)".r
+      final private val latestSbtVersions = List(sbt.defaultSbtVersion, sbt.latestSbtVersion_0_13)
+        .map(_.value)
+
+      def proposeBuildPropertiesUpdate(): F[List[Update.Single]] =
```

Comment:
I think this should return `F[Option[Update.Single]]` to make clear that it returns at most one element. At the usage site it can then be converted to a list for easier concatenation with the other updates.

Reply:
I've another suggestion. If you extract the part of this function that transforms the content of `build.properties` to an update (`String => Option[Update.Single]`), that function has no effect in `F[_]` and can be tested more easily.

Reply:
Mad it `Option` in https://github.com/fthomas/scala-steward/pull/665/commits/2a407f257f2629154d9cae4f490af68a81cd437c
Extracted the method in https://github.com/fthomas/scala-steward/pull/665/commits/a3516c8edebf20a9e8821586905af4287bf9e7fe